### DISCUSSION TITLE
Limiting based on association

### DISCUSSION THREAD

**Discussion by jessehanley** [Shopify/limiter#30] [Q&A] [answered]
Hey Shopify crew! 

Thanks for making such amazing gems. 

Use so many of them in my products, can't tell you how much I appreciate it.

Quick q: if I'd like to limit based on an association count, what would be the best way to handle it using this library? Say, I want to limit the intake of `event`s being created per visitor (i.e 100 events per visitor). 

Thank you! ⭐

**Comment by sbfaulkner** [comment, marked as answer]
@jessehanley this gem is for rate-limiting calls to methods, not for any form of activemodel vallidation.

**Comment by realleoman** [comment]
Hi @sbfaulkner  This gem is amazing. I was able to limit the calls to my endpoint but for some reason when I add the block after the limit_method, I am not able to make it work, not even having `puts` within. Any ideas why? 

I need the API to get the token authenticated before calling  the endpoint

```
  before_action :http_auth_header, only: :orders_history

  limit_method(:orders_history, rate: 1, interval: 60) do
    waiting_time
  end

  def waiting_time
    render json: { error: 'Rate Limit Exceeded. Please wait a minute to try again'}, status: :bandwidth_limit_exceeded  and return
  end

  def orders_history
     puts "I accessed orders_history at #{Time.zone.now}"
     ....
 end

 ```
